pkg/compat/client.go

```diff
@@ -109,11 +122,9 @@ func (c Client) Get(ctx context.Context, key k8sclient.ObjectKey, in runtime.Obj
 	if err != nil {
 		return err
 	}
-	if in != obj {
```

Comment:
The purpose for this check is so that we don't up-convert the result unless it was down-converted.

Reply:
I think it is already being done in the scheme.Convert

Reply:
Drilling down in the code, it seems it's still doing _some_ work we could/should avoid.  Thoughts?

Reply:
Agreed